src/Mod/Sketcher/App/Constraint.cpp

```diff
@@ -322,6 +329,23 @@ bool Constraint::hasElement(int index) const
     return index >= 0 && index < elements.size();
 }
 
+bool Constraint::isElementsEmpty() const
+{
+    return elements.empty();
+}
+
+void Constraint::truncateElements(size_t newSize)
+{
+    if (newSize < elements.size()) {
+        elements.resize(newSize);
+    }
+}
+
+void Constraint::pushBackElement(GeoElementId elt)
+{
+    elements.push_back(elt);
+}
+

```

Comment:
Shouldn't be in this commit but in [Sketcher: Constraint: Replace 'First' 'FirsPos' 'Second'... by vector](https://github.com/FreeCAD/FreeCAD/pull/22217/commits/61aeb765cc5503d25db21dbdc6a66ac325821ec6), yes?

Reply:
This PR has 2 commits, the one from that other PR, and the text tool.
So the other PR has to merge first and this PR rebased. Unless all is merged here.

Reply:
But this code is in https://github.com/FreeCAD/FreeCAD/pull/22217/commits/711c7cfad80842dc79f524e2a76b4547ed508d7e

Reply:
Ah yes it's because I needed more functions for the next constraints. Well arguably it can be in the second commit :)